### ISSUE TITLE
CTRL+ALT+SHIFT combination don't work in Linux Mint 19

### ISSUE THREAD

**Issue by artcer** [linuxmint/mintsystem#85]
I have two computers: the first one is laptop which previously had Linux Mint 18.3 and I upgraded it to Linux Mint 19. The second has fresh Linux Mint 19 install.

The problem is with ALT key in combination "CTRL+ALT+SHIFT". It seems that Linux Mint 19 doesn't recognize ALT key in this combination, even though in Linux Mint 18.3 it used to work perfectly. Some examples: in Linux Mint 19 combination CTRL+ALT+T works fine (opens Terminal), CTRL+ALT+LEFT ARROW (or RIGHT ARROW) works fine (changes workspace), ALT+F4 works fine (closes active window). But combination CTRL+ALT+SHIFT+LEFT ARROW (or RIGHT ARROW) should move active window to another workspace, but it doesn't even though in system there is set a shortcut for that.

I tested it with Intellij which has the ability to check what key combination is pressed. And when I press CTRL+ALT+SHIFT combination, Intellij recognizes it as CTRL+SHIFT without ALT. It also happens when in Linux Mint 19 system settings for shortcuts I try to change any shortcut combination. For example: Keyboard -> Shortcuts -> Windows -> Inter-workspace there is and option to set shortcut for "Move window to workspace 1". If I try to set a shortcut for this action as CTRL+ALT+SHIFT+Num Pad 1, newly assigned shortcut is CTRL+ALT+1. When I try to change shortcut for "Move window to right workspace" and press CTRL+ALT+SHIFT+RIGHT ARROW, newly set shortcut is set as CTRL+SHIFT+RIGHT ARROW.

So basically combination of CTRL+ALT+SHIFT is no longer possible in Linux Mint 19.

**Comment by icarter09** [comment]
Cinnamon 3.8.9
Linux Mint 19 64bit

I'm not able to duplicate this issue. After reading this ticket, I went to "Keyboard" and saw that the 'Move window to right workspace' and 'Move window to left workspace' have the keyboard shortcuts as 'Shift+Ctrl+Alt+Right (or Left for the given workspace) by default. Verified the window is moved when the keyboard shortcut is applied. 

@artcer are you still having this issue?

**Comment by acernauskas** [comment]
@icarter09 I still have this issue. 

For testing purposes I installed [screenkey](https://github.com/wavexx/screenkey) to see on the screen key combinations when I press keys. This is my testing results:

1. CTRL + ALT + -> shows correctly
2. CTRL + ALT + <- shows correctly
3. CTRL + SHIFT + -> shows correctly
4. CTRL + SHIFT + <- shows correctly
5. SHIFT + ALT shows as ISO_Next_Group which I guess is not correct. This combination (shift+alt changes my keyboard layout from english to lithuanian and in Keyboard -> Layouts -> Options -> Switching to another layout I have selected Alt+Shift option)
6. CTRL + SHIFT + ALT + -> shows various combinations depending which key was pressed first, second and third in this combination, for example shows CTRL + ISO_Next_Group or CTRL + ALT + ISO_Next_Group or ALT + ISO_Next_Group and so on. So in this case this is also not correct.

So maybe CTRL + SHIFT + ALT combination not working properly because of this ISO_Next_Group?

**Comment by artcer** [comment]
@icarter09 I still have this issue. 

For testing purposes I installed [screenkey](https://github.com/wavexx/screenkey) to see on the screen key combinations when I press keys. This is my testing results:

1. CTRL + ALT + -> shows correctly
2. CTRL + ALT + <- shows correctly
3. CTRL + SHIFT + -> shows correctly
4. CTRL + SHIFT + <- shows correctly
5. SHIFT + ALT shows as ISO_Next_Group which I guess is not correct. This combination (shift+alt changes my keyboard layout from english to lithuanian and in Keyboard -> Layouts -> Options -> Switching to another layout I have selected Alt+Shift option)
6. CTRL + SHIFT + ALT + -> shows various combinations depending which key was pressed first, second and third in this combination, for example shows CTRL + ISO_Next_Group or CTRL + ALT + ISO_Next_Group or ALT + ISO_Next_Group and so on. So in this case this is also not correct.

So maybe CTRL + SHIFT + ALT combination not working properly because of this ISO_Next_Group?

**Comment by mtwebster** [comment]
Next group is probably the layout switcher key - keyboard settings->layouts, then "Options" - you can select a different combination.  NOTE: You must log out/back in for this change to take effect (this is out of our control).

**Comment by artcer** [comment]
@mtwebster yes, changing layout switcher key in "Switching to another layout" to any other combination allows CTRL+ALT+SHIFT+Arrow move currently active window to another workspace. Is there any way I can keep SHIFT+ALT as my layout switcher key and at the same time CTRL+SHIFT+ALT+Arrow as key combination for moving currently active window to another workspace? Until LM 19 these settings didn't cross and worked perfectly but it seems that in LM 19 they cannot work together. I would like to keep these key combinations and for LM 19 as I was using them in past LM versions.

**Comment by mtwebster** [comment]
It's an upstream issue - check out the same discussion here: https://github.com/linuxmint/Cinnamon/issues/7886

It was always patched to our old, 'good' behavior but I think since gnome-shell doesn't depend on it, and they dumped Unity, they quit patching it.
